tests/unit/FileReader/ReaderFactoryTest.php

```diff
@@ -24,25 +25,46 @@ public function setUp()
         $this->factory = new ReaderFactory([
             '.php'  => PHPFileReader::class,
             '.json' => JSONFileReader::class,
+            '.yaml' => YAMLFileReader::class,
+            '.yml' => YAMLFileReader::class,
         ]);
     }
 
-    public function testCreatesAReader()
+    /**
+     * @dataProvider providerCreatesAppropriateFileReader
+     *
+     * @param string $extension
+     * @param string $fileReaderClass
+     */
+    public function testCreatesAppropriateFileReader($extension, $fileReaderClass)
     {
-        $this->createTestFile('test.php');
+        $filename = 'test' . $extension;
 
-        $reader = $this->factory->create($this->getTestPath('test.php'));
+        $this->createTestFile($filename);
 
-        $this->assertInstanceOf(PHPFileReader::class, $reader);
+        $reader = $this->factory->create($this->getTestPath($filename));
+
+        $this->assertInstanceOf($fileReaderClass, $reader);
     }
 
-    public function testCreatesAnotherReader()
+    /**
+     * @return \Generator
+     */
+    public function providerCreatesAppropriateFileReader()
```

Comment:
@tomphp 

Since I would have otherwise used a data provider to provide the different extensions `.yaml` and `.yml`, I figured it would make sense to use a data provider altogether, what do you think?